app/models/manageiq/providers/hawkular/middleware_manager/live_metrics_capture.rb

```diff
@@ -61,27 +74,21 @@ def min_max_timestamps(client, metric_id)
     end
 
     def fetch_metrics(metric_id, metric_type, starts, ends, bucket_duration)
-      case metric_type
-      when "GAUGE"        then gauges(metric_id, starts, ends, bucket_duration)
-      when "COUNTER"      then counters(metric_id, starts, ends, bucket_duration)
-      when "AVAILABILITY" then availabilities(metric_id, starts, ends, bucket_duration)
-      else raise MetricValidationError, "Validation error: unknown type #{metric_type}"
-      end
-    end
-
-    def gauges(metric_id, starts, ends, bucket_duration)
-      sort_and_normalize(@gauges.get_data(metric_id, :starts => starts, :ends => ends,
-                                          :bucketDuration => bucket_duration))
-    end
-
-    def counters(metric_id, starts, ends, bucket_duration)
-      sort_and_normalize(@counters.get_rate(metric_id, :starts => starts, :ends => ends,
-                                          :bucket_duration => bucket_duration))
+      sort_and_normalize(fetch_raw_metrics(metric_id, metric_type, starts, ends, bucket_duration))
     end
 
-    def availabilities(metric_id, starts, ends, bucket_duration)
-      sort_and_normalize(@avail.get_data(metric_id, :starts => starts, :ends => ends,
-                                         :bucketDuration => bucket_duration))
+    def fetch_raw_metrics(metric_id, metric_type, starts, ends, bucket_duration)
+      data = case metric_type
+             when "GAUGE"
+               @gauges.get_data(metric_id, :starts => starts, :ends => ends, :bucketDuration => bucket_duration)
+             when "COUNTER"
+               @counters.get_rate(metric_id, :starts => starts, :ends => ends, :bucket_duration => bucket_duration)
+             when "AVAILABILITY"
+               @avail.get_data(metric_id, :starts => starts, :ends => ends, :bucketDuration => bucket_duration)
+             else
+               raise MetricValidationError, "Validation error: unknown type #{metric_type}"
```

Comment:
Perhaps this MetricValidationError has not sense once these classes have evolved.
metrics type are controlled in the implementation as it is not make sense to raise this one.